torch/csrc/jit/codegen/cuda/scheduler/registry.cpp

```diff
@@ -264,6 +267,214 @@ class SchedulerTopologyChecker {
+}
+
+size_t SchedulerRuntimeInfo::getAlignmentSize(TensorView* tv) {
+  auto alignment_entry = alignment_map_.find(tv);
+  if (alignment_entry == alignment_map_.end()) {
+    return max_alignment_size_in_byte;
+  } else {
+    return alignment_entry->second;
+  }
+}
+
+void SchedulerRuntimeInfo::initializeExpressionEvaluator(
+    const at::ArrayRef<IValue>& inputs) {
+  // TODO: refactor bindFusionInputs to better support this
+  //  use case, i.e. support construct and bind input.
+  expression_evaluator_ =
+      std::make_unique<ExpressionEvaluator>(complete_fusion_);
+  *expression_evaluator_ =
+      executor_utils::bindFusionInputs(inputs, complete_fusion_);
+}
+
+size_t SchedulerRuntimeInfo::collectAlignmentSize(
+    const at::Tensor& tensor) const {
+  const size_t address = reinterpret_cast<size_t>(tensor.data_ptr());
+  size_t alignment_size = 1;
+  size_t next_alignment_size = 2;
+
+  while (alignment_size <= max_alignment_size_in_byte &&
+         address % next_alignment_size == 0) {
+    alignment_size = next_alignment_size;
+    next_alignment_size *= 2;
+  }
+
+  return alignment_size;
+}
+
+void SchedulerRuntimeInfo::collectVectorizationInfo(
+    const at::ArrayRef<IValue>& inputs) {
+  common_alignment_size_ = max_alignment_size_in_byte;
+  size_t number_of_inputs = complete_fusion_->inputs().size();
+  std::unordered_map<TensorView*, size_t> cg_tensor_to_at_tensor_index;
+
+  for (auto input_index : c10::irange(number_of_inputs)) {
+    if (auto input_tensor = dynamic_cast<TensorView*>(
+            complete_fusion_->inputs()[input_index])) {
+      if (input_tensor->nDims() == 0) {
+        // A 0-dim tensor input would not need vectorization
+        continue;
+      }
+      if (input_tensor->domain()
+              ->domain()[input_tensor->nDims() - 1]
+              ->isBroadcast()) {
+        // skip the tensors with innermost iterdomain broadcasted,
+        //  as we will not vectorize these.
+        continue;
+      }
+
+      // Collect strides of the input tensor
+      TORCH_INTERNAL_ASSERT(inputs[input_index].isTensor());
+      const auto& at_tensor = inputs[input_index].toTensor();
+
+      cg_tensor_to_at_tensor_index.emplace(
+          std::make_pair(input_tensor, input_index));
+
+      // Collect alignment of the input tensor
+      auto alignment_size = collectAlignmentSize(at_tensor);
+      common_alignment_size_ = std::min(alignment_size, common_alignment_size_);
+      alignment_map_[input_tensor] = alignment_size;
+    }
+  }
+
+  // Compute max vector word size for each input,
+  //  tensors with inner most broadcast already
+  //  filtered out.  common_alignment_size_ is
+  //  computed up to this point.
+  for (auto it : cg_tensor_to_at_tensor_index) {
+    vectorword_map_[it.first] = collectMaxVectorizeSize(
+        inputs[it.second].toTensor(), common_alignment_size_);
+  }
+}
+
+size_t SchedulerRuntimeInfo::collectMaxVectorizeSize(
```

Comment:
@csarofeen 
This is function for collecting max vector size (in items) for input tensors as we discussed. Please let me know if this makes sense and if more info is needed.